parlai/agents/fid/fid.py

```diff
@@ -376,6 +376,12 @@ def _set_query_vec(self, observation: Message) -> Message:
         self.show_observation_to_echo_retriever(observation)
         super()._set_query_vec(observation)
 
+    def eval_step(self, batch):
+        output = super().eval_step(batch)
+        if hasattr(self.model_api.retriever, 'clear_mapping'):
```

Comment:
This looks like a good solution. But what about cleaning on batchify that you suggested?

Reply:
I realize I can't do that in batchify, since model act flow is observe (insert k,v pairs to the `_saved_doc` and `_query_ids`) -> batchify -> act;
if I clear up the mapping during batchify then that literally clear up all the newly inserted kv pairs that are from the same batch.

Also notice another problem with this change (it won't reset the mapping in train step), let me update the pr and ping you guys again after it's ready.

Reply:
What if we clear after act/batch_act from the FiD agent?

Reply:
yea that's my sol rn.